go/coredns/plugin/pfdns/pfdns.go

```diff
@@ -122,21 +122,22 @@ func (pf *pfdns) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 		if k.MatchString(state.QName()) {
 			answer, err := pf.LocalResolver(state)
 			if err != nil {
-				fmt.Println("Local resolver error with the following error" + err.Error())
+				fmt.Println("Local resolver error for fqdn" + state.QName() + " with the following error" + err.Error())
 			} else {
 				for _, ans := range append(answer.Answer, answer.Extra...) {
 					switch ansb := ans.(type) {
 					case *dns.A:
 						for _, valeur := range v {
 							if err := pf.SetPassthrough(ctx, "passthrough", ansb.A.String(), valeur, true); err != nil {
-								fmt.Println("Not able to contact localhost", err)
+								fmt.Println("Not able to contact localhost for setPassthrough", err)
 							}
 						}
 					}
 				}
+				fmt.Println(srcIP + " : " + mac + " Domain bypass for fqdn " + state.QName())
+				w.WriteMsg(answer)
 			}
-			fmt.Println(srcIP + ":" + mac + " Domain bypass for fqdn " + state.QName())
-			return pf.Next.ServeDNS(ctx, w, r)
+			return 0, nil
```

Comment:
It will never be executed

Reply:
It is getting executed in my code `return 0, nil` on line 140, 166, 192 is shared between the `if` and `else` codes.  On my tests I am seeing being called on both the positive and negative cases in the if. 

The `w.WriteMsg(answer)` requires also a `return 0, nil` so my idea was to share it between the positive and negative case

Reply:
Yes make sense.